src/ppx/grfx/vk/vk_command.cpp

```diff
@@ -290,36 +290,63 @@ void CommandBuffer::ClearRenderTarget(
     grfx::Image*                        pImage,
     const grfx::RenderTargetClearValue& clearValue)
 {
-    auto pCurrentRenderPass = GetCurrentRenderPass();
-    if (IsNull(pCurrentRenderPass)) {
+    if (!HasActiveRenderPass()) {
         return;
     }
 
-    // Make sure pImage is a render target in current render pass
-    const uint32_t renderTargetIndex = pCurrentRenderPass->GetRenderTargetImageIndex(pImage);
-    if (renderTargetIndex == UINT32_MAX) {
-        return;
+    grfx::Rect renderArea;
+    uint32_t   colorAttachment = UINT32_MAX;
+    uint32_t   baseArrayLayer;
+
+    // Dynamic render pass
+    if (mDynamicRenderPassActive) {
+        renderArea = mRenderArea;
+
+        auto views = mRenderTargetViews;
+        for (uint32_t i = 0; i < views.size(); ++i) {
+            auto rtv   = views[i];
+            auto image = rtv->GetImage();
+            if (image.Get() == pImage) {
+                colorAttachment = i;
+                baseArrayLayer  = rtv->GetArrayLayer();
+                break;
+            }
+        }
     }
+    else {
+        // active regular render pass
+        auto pCurrentRenderPass = GetCurrentRenderPass();
+        renderArea              = pCurrentRenderPass->GetRenderArea();
+
+        // Make sure pImage is a render target in current render pass
+        const uint32_t renderTargetIndex = pCurrentRenderPass->GetRenderTargetImageIndex(pImage);
+        colorAttachment                  = renderTargetIndex;
 
-    // Get view at renderTargetIndex
-    auto view = pCurrentRenderPass->GetRenderTargetView(renderTargetIndex);
+        // Get view at renderTargetIndex
+        auto view      = pCurrentRenderPass->GetRenderTargetView(renderTargetIndex);
+        baseArrayLayer = view->GetArrayLayer();
+    }
+
+    if (colorAttachment == UINT32_MAX) {
+        PPX_LOG_WARN("pImage is not a render target.");
```

Comment:
Shall this be a warn or an assert?
Could that happen without it being an actual bug?

Reply:
I think technically it should be an assert. It would happen if someone passes an image that is not part of the renderpass, in which case it wouldn't work, so it won't do the clear. I don't see a reason why anyone would want to do that and not know that the clear did not work. So probably an assert is warranted.